datadog_lambda/trigger.py

```diff
@@ -369,3 +367,28 @@ def extract_http_status_code_tag(trigger_tags, response):
         status_code = response.status_code
 
     return str(status_code)
+
+
+def is_step_function_event(event):
+    """
+    Check if the event is a step function that invoked the current lambda.
+
+    The whole event can be wrapped in "Payload" in Legacy Lambda cases. There may also be a
+    "_datadog" for JSONata style context propagation.
+
+    The actual event must contain "Execution", "StateMachine", and "State" fields.
+    """
+    event = event.get("Payload", event)
+
+    # JSONPath style
+    if all(field in event for field in ("Execution", "StateMachine", "State")):
```

Comment:
I hate to say it (cuz this code is so easy to read and maintain as it is) that using `all` and the tuple are probably slower than just doing it manually. For the way it's written now, you gotta allocate the tuple, plus create the generator, all before even doing any work to evaluate the `all`.

I'd be curious to know the timing difference though. I'd recommend using the `timeit` module, or just running some benchmarks manually.

Here's what I'm thinking for the naive approach to compare this to:

```py
if "Execution" in event and "StateMachine" in event and "State" in event:
    return True
```

Reply:
Did a quick benchmark and looks like using `all()` is >4x slower 😮

I'll switch it back to simple approach